src/Translation_unit.ml

```diff
@@ -144,24 +144,39 @@ let parse_print (XUnit xunit) (conf : Conf.t) ~input_name ~input_file ic
   (* iterate until formatting stabilizes *)
   let rec print_check ~i ~(conf : Conf.t) ~ast ~comments ~source_txt
       ~source_file ~postprocess : result =
-    let tmp, oc =
-      if not Conf.debug then Filename.open_temp_file ~temp_dir:dir base ext
+    dump xunit dir base ".old" ".ast" ast ;
+    let remaining_comments, tmp =
+      let print_and_get_remaining_comments wrap oc =
+        let source = Source.create source_txt in
+        let cmts_t = xunit.init_cmts source conf ast comments in
+        let fs = Format.formatter_of_out_channel oc in
+        postprocess fs ;
+        Fmt.set_margin conf.margin fs ;
+        wrap (xunit.fmt source cmts_t conf ast) fs ;
```

Comment:
`wrap` is not very informative, and is used elsewhere, what about `maybe_debug` or something? And it is probably clearer to just have a named bool arg instead of passing either Fn.id or Fmt.with_box_debug around.

Reply:
I agree, I changed to use a boolean parameter.